sdk/keyvault/keyvault-certificates/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -93,35 +128,37 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
       throw new Error("The resource address for authorization must use the 'https' protocol.");
     }
 
-    let originalBody = webResource.body;
+    const originalBody = webResource.body;
 
-    if (this.challenge == undefined) {
+    if (this.challengeCache.challenge == undefined) {
       // Use a blank to start the challenge
       webResource.body = "";
     } else {
       // or use the cached token if we have one
       await this.authenticateRequest(webResource);
     }
 
-    let response = await this._nextPolicy.sendRequest(webResource);
+    const response = await this._nextPolicy.sendRequest(webResource);
 
     if (response.status == 401) {
       webResource.body = originalBody;
 
       let www_authenticate = response.headers.get("WWW-Authenticate");
 
       if (www_authenticate) {
-        let resource = this.parseWWWAuthenticate(www_authenticate);
-        let challenge = new AuthenticationChallenge(resource + "/.default")
+        const { authorization, resource } = this.parseWWWAuthenticate(www_authenticate);
+        const challenge = new AuthenticationChallenge(authorization, resource + "/.default")
 
-        if (this.challenge != challenge) {
-          this.challenge = challenge;
+        if (!challenge.equalTo(this.challengeCache.challenge)) {
```

Comment:
ditto as comment above about cache lookup

Reply:
There's only one challenge we're caching. The reason we're "caching" it is because the class gets re-generated on every request. There's no multitude of challenges we will be receiving in these pipelines, as far as I'm aware.